src/jit/codegenlinear.cpp

```diff
@@ -113,6 +113,10 @@ void CodeGen::genInitialize()
     // Make sure a set is allocated for compiler->compCurLife (in the long case), so we can set it to empty without
     // allocation at the start of each basic block.
     VarSetOps::AssignNoCopy(compiler, compiler->compCurLife, VarSetOps::MakeEmpty(compiler));
+
+    // We initialize the stack level before fist "BasicBlock" code is generated in case we need to report stack
```

Comment:
typo: before fist

Reply:
> we need to report stack variable needs home and so its stack offset.

In which function do we need it/access StackLevel value before `genCodeForBBlist`?

Reply:
At the beginning of the for there are two calls before [those lines](https://github.com/BrianBohe/coreclr/blob/brbohe/moving_genStackLevel_to_codeGenInterface/src/jit/codegenlinear.cpp#L204-L206):

1. **recordVarLocationsAtStartOfBB(block)**: which is able to report VariableLiveRanges if the home is different than what it on on LclVarDsc.
2. : **genUpdateLife(block->bbLiveIn)**: This update GC references, registers masks, and in my pr report variable ranges. You will came out with a strange offset for the variables that are alive on the "block->bbLiveIn". This would happen in one of the crossgen tests. I can't tell you exactly file/method is but I if you dump the jitted methods this is the case "Compiling   56 System.ArgIterator::.ctor, IL size = 15, hsh=0x865b7588".